src/nb_func.cpp

```diff
@@ -445,7 +445,8 @@ PyObject *nb_func_new(const void *in_) noexcept {
                 // skips Python's usual logic where __init__ is always called
                 // if __new__ returns an instance of the type.
                 bool noargs_ok = true;
-                for (size_t i = 1; i < fc->nargs - has_var_kwargs; ++i) {
+                for (size_t i = 1;
+                     i < fc->nargs - static_cast<size_t>(has_var_kwargs); ++i) {
```

Comment:
Since `int` can represent the entire range of values of the original types on the right, integer promotion converts each value to `int`.  Then subtraction results in an `int`.   So, may I suggest:
```
    for (int i = 1; i < fc->nargs - has_var_kwargs; ++i) {
```
I feel this is more natural/readable (and it fits on one line).  As a bonus, the generated code might be better.  On Compiler Explorer, I think it's nicer to uncheck the "Intel asm syntax" box.  Note that the `int` code is smaller using 32-bit integers than for 64-bit size_t, because `addq` takes one more byte than `addl`, etc.

  64bit: `48 83 c3 01  add $0x1,%rbx`    ==>  32bit: `83 c3 01  add $0x1,%ebx`
  64bit: `48 39 eb     cmp %rbp,%rbx`  ==>  32bit: `39 eb     cmp %ebp,%ebx`